src/strategies/spidStrategy.ts

```diff
@@ -129,6 +130,61 @@ const spidStrategy = (
     }
   };
 
+  const idpMetadata = await parseIdpMetadata(); // FIXME: handle the exception and use default configuration
+  for (const idp of idpMetadata) {
+    const idpOption = Object.assign({}, idp);
+    delete idpOption.entityID; // TODO: no-object-mutation
+    switch (idp.entityID) {
+      case "https://loginspid.aruba.it":
```

Comment:
you can use a single mapping function that relies on a map between IdP url and ID:

```ts
const idpIdForUrl = {
  "https://loginspid.aruba.it": "arubaid"
  // ...
};
```

Reply:
fix on commit d874196